### PR TITLE
feat(storage): rename tables

### PR DESCRIPTION
**Pull request by shekhirin**
Work on top of https://github.com/paradigmxyz/reth/pull/6062, it had too many conflicts, and it was easier to do a find-and-replace all over again. h/t to @JosepBove for the initial impl.

Tables https://github.com/paradigmxyz/reth/blob/f8dd4a16ebd366e485883c615adc738899fca560/crates/storage/db/src/tables/mod.rs#L244-L375

### CODE REVIEW

**Review by joshieDo — APPROVED** [review]

**Review by onbjerg — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

File: crates/stages/src/stages/sender_recovery.rs [hunk 1]
```diff
@@ -455,10 +457,11 @@ mod tests {
...
         /// # Panics
         ///
-        /// 1. If there are any entries in the [tables::TxSenders] table above a given block number.
+        /// 1. If there are any entries in the [tables::TransactionSenders] table above a given
+        /// block number.
         ///
-        /// 2. If the is no requested block entry in the bodies table, but [tables::TxSenders] is
-        ///    not empty.
+        /// 2. If the is no requested block entry in the bodies table, but
+        /// [tables::TransactionSenders] is    not empty.
```
Comment (emhane) [inline crates/stages/src/stages/sender_recovery.rs:464]:
```suggestion
        /// [tables::TransactionSenders] is not empty.
```

File: crates/storage/provider/src/providers/database/metrics.rs [hunk 2]
```diff
@@ -50,14 +50,14 @@ pub(crate) enum Action {
...
     InsertHeaders,
     InsertHeaderNumbers,
-    InsertHeaderTD,
+    InsertHeaderTerminalDifficulties,
     InsertBlockOmmers,
-    InsertTxSenders,
+    InsertTransactionSenders,
     InsertTransactions,
-    InsertTxHashNumbers,
+    InsertTransactionHashNumberss,
```
Comment (emhane) [inline crates/storage/provider/src/providers/database/metrics.rs:57]:
```suggestion
    InsertTransactionHashNumbers,
```

File: crates/stages/src/stages/sender_recovery.rs [hunk 3]
```diff
@@ -53,7 +53,7 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
     /// [`BlockBodyIndices`][reth_db::tables::BlockBodyIndices],
     /// collect transactions within that range,
     /// recover signer for each transaction and store entries in
```
Comment (emhane) [inline crates/stages/src/stages/sender_recovery.rs:55]:
```suggestion
    /// collect transactions within that range, recover signer
    /// for each transaction and store entries in
```

**Review by emhane — CHANGES_REQUESTED** [review]
super clean, very helpful descriptive names. a lot of work though, this is why it's nice to just do proper names from the start.

comments (not caught by rustdocs) seem to check out with new naming, good!

**Review by emhane — APPROVED** [review]

**Pull request merged**